lib/backend/k8s/syncer.go

```diff
@@ -325,9 +354,9 @@ func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[model.Key]bool, re
 			labels.Revision = profile.Revision
 
 			snap = append(snap, *rules, *tags, *labels)
-			keys[rules.Key] = true
-			keys[tags.Key] = true
-			keys[labels.Key] = true
+			keys[fmt.Sprintf("%s", rules.Key)] = true
```

Comment:
Yeah, can do.  I don't really like this change at all.

I'd much prefer to make the requirement that a Key needs to be hashable, but IPPoolKey has a net.IPNet in it which breaks that.

We could tweak to use a pointer to a CIDR instead, but I that wouldn't work since two equal keys wouldn't hash to the same thing.

Maybe we could implement a hash method or something to make it hashable? I can't seem to find any examples of that.